File: challenge_1/javascript_tiagox/change.js
function giveMeChange(coins, change) {
    let coin = coins.slice(-1).shift();
    if (coin) {
        return Math.floor(change / coin) + gimeMeChange(coins.slice(0, -1), change % coin);
    } else {
        return 0;
    }
}

console.log(giveMeChange([1, 5, 10], 28)); // => 6

File: challenge_1/java_juan/Change.java
public class Change {

	public static void main(String[] args) {
		int[] values = {1, 5, 10};
		System.out.println(giveMeChange(values, 28));
	}
	
	public static int giveMeChange(int[] values, int change){
		 int numberOfCoins = 0;

		 if(values.length>0){
			 for(int index = values.length; index>0; index--){
				numberOfCoins += change / values[index-1];
			    change = change % values[index-1];
			 }	 
		 }
		 
		 return numberOfCoins;
	}
}
